### ISSUE TITLE
ashm_matrix

### ISSUE THREAD

**Issue by bbai89** [morinlab/GAMBLR.predict#1]
I would like to try classify_fl on our dataset. But I realised that the ashm_matrix created by classify_fl is not correct. For example the mutation at 1_27023597 is assigned as BTG2_intronic and ITPKB_TSS. If I look at "ashm_features_bed", the start and end of BTG2_intronic is 203274698 and 203275778, the start and end of ITPKB_TSS is 226920563 and 226927885. 1_27023597 does not fall into any of these regions, why it was assigned with two features. Please help. Thank you very much!

Baoyan

**Comment by Kdreval** [comment]
Hi @bbai89 , I will be happy to help with this. To make sure we are on the same page, can you please show how you called the classify_fl() function and provide the maf entry for the variant in question, at least with the columns Chromosome, Start_Position, End_Position, and Variant_Classification?

**Comment by bbai89** [comment]
Thank you for the quick reply. The command to call classify_fl

pre<-classify_fl(
  these_samples_metadata = meta_pre,
  maf_data = flreseq_maf_pre,
  output = "both")


and here are the joined column after GAMBLR.helpers::cool_overlaps

Hugo_Symbol	Entrez_Gene_Id	Center	NCBI_Build	Chromosome	Start_Position	End_Position	Strand	Variant_Classification	Variant_Type	Reference_Allele	Tumor_Seq_Allele1	Tumor_Seq_Allele2	dbSNP_RS	dbSNP_Val_Status	Tumor_Sample_Barcode	Matched_Norm_Sample_Barcode	Match_Norm_Seq_Allele1	Match_Norm_Seq_Allele2	Tumor_Validation_Allele1	Tumor_Validation_Allele2	Match_Norm_Validation_Allele1	Match_Norm_Validation_Allele2	Verification_Status	Validation_Status	Mutation_Status	Sequencing_Phase	Sequence_Source	Validation_Method	Score	BAM_File	Sequencer	Tumor_Sample_UUID	Matched_Norm_Sample_UUID	HGVSc	HGVSp	HGVSp_Short	Transcript_ID	Exon_Number	t_depth	t_ref_count	t_alt_count	n_depth	n_ref_count	n_alt_count	start	end	name
4856	ARID1A	0	.	GRCh37	1	27023597	27023597	+	Frame_Shift_Del	deletion	AGAGGTGGCACTCC	AGAGGTGGCACTCC	A	novel		P4_1		AGAGGTGGCACTCC	AGAGGTGGCACTCC			AGAGGTGGCACTCC	AGAGGTGGCACTCC												c.705_717del	p.G236fs	p.G236fs	ENST00000324856.7	exon1	974	775.9858	198.0142	960	974	0	203274698	203275778	BTG2_intronic
4857	ARID1A	0	.	GRCh37	1	27023597	27023597	+	Frame_Shift_Del	deletion	AGAGGTGGCACTCC	AGAGGTGGCACTCC	A	novel		P4_1		AGAGGTGGCACTCC	AGAGGTGGCACTCC			AGAGGTGGCACTCC	AGAGGTGGCACTCC												c.705_717del	p.G236fs	p.G236fs	ENST00000324856.7	exon1	974	775.9858	198.0142	960	974	0	226920563	226927885	ITPKB_TSS
5007	ARID1A	0	.	GRCh37	1	27023597	27023597	+	Frame_Shift_Del	deletion	AGAGGTGGCACTCC	AGAGGTGGCACTCC	A	novel		P4_2		AGAGGTGGCACTCC	AGAGGTGGCACTCC			AGAGGTGGCACTCC	AGAGGTGGCACTCC												c.705_717del	p.G236fs	p.G236fs	ENST00000324856.7	exon1	838	595.9856	242.0144	960	838	0	203274698	203275778	BTG2_intronic
5008	ARID1A	0	.	GRCh37	1	27023597	27023597	+	Frame_Shift_Del	deletion	AGAGGTGGCACTCC	AGAGGTGGCACTCC	A	novel		P4_2		AGAGGTGGCACTCC	AGAGGTGGCACTCC			AGAGGTGGCACTCC	AGAGGTGGCACTCC												c.705_717del	p.G236fs	p.G236fs	ENST00000324856.7	exon1	838	595.9856	242.0144	960	838	0	226920563	226927885	ITPKB_TSS


let me know if you want to have a text file. 
Thanks again!
Baoyan

**Comment by Kdreval** [comment]
Thanks for the provided info!
I tried to replicate this error but the underlying overlap function works as expected, not annotating this position as aSHM in any region. I tried both the data you provided and a toy position adjacent tho the variant you noted and it still works correctly.
One thing I noticed is that in the data you provided few columns seem to be out of order - for example, the `Chromosome` is in 5th position in header but 6th position in data (for each of the 4 rows). Is it possible that there is a mismatch between columns and their values in the incoming maf?
To narrow down whether this is issue related to functionality or the data itself, can you please post the output you get when running this? 
```
library(GAMBLR.helpers)
library(dplyr)


test_maf <- data.frame(
    Chromosome = c("1", "1", "1"),
    Start_Position = c(27023597, 203274698, 203275778),
    End_Position = c(27023597, 203274698, 203275778),
    Annotation = c("No_mapping", "Maps_to_BTG2", "Maps_to_ITPKB")
)

ashm_features_bed <- data.frame(
    chrom = c("1", "1"),
    start = c(203274698, 226920563),
    end = c(203275778, 226927885),
    name = c("BTG2_intronic", "ITPKB_TSS")
)

cool_overlaps(
    data1 = test_maf,
    data2 = ashm_features_bed,
    columns2 = c("chrom", "start", "end")
)
```
Expecting it to return 2 rows that properly match and the variant in question to be omitted from the output (since it does not match) - but wonder if it works the same on your side.

**Comment by bbai89** [comment]
Hi,

Thank you for testing it the weekend. I tried your suggestion. It worked as
expected.

I realized that End_position is  character (sorry!). After resetting them
as numeric, the overlap function worked. Thank you very much for your time
and instant response.

Kind Regards,
Baoyan

**Comment by Kdreval** [comment]
Always glad to help!
Thank you for giving this a try.
Please do not hesitate to report more issues or ask questions if you have any.

**Comment by Kdreval** [comment]
I also thought it might be helpful to control for this scenario, so[ the recent update](https://github.com/morinlab/GAMBLR.predict/commit/1918afa6ada6d3c7aac67fa3c720cc9cace84496) to master should handle such cases.

**Comment by bbai89** [comment]
Just a quick question: is it possible to change parameters in RFmodel_FL? I
read somewhere that this is an option only for developers.

**Comment by Kdreval** [comment]
You can provide different RF model (constructed with RandomForest package) to use with this function by passing it with `model` argument: https://github.com/morinlab/GAMBLR.predict/blob/1918afa6ada6d3c7aac67fa3c720cc9cace84496/R/classifiers.R#L15-L17
This is just a convenience functionality to test performance when different classifiers are used. The discouraging note is there because the default model is the one used in the [Blood paper](https://pubmed.ncbi.nlm.nih.gov/37084389/), so when different model is supplied it might not give the same prediction as in the paper.

**Comment by bbai89** [comment]
Thank you very much for explaining it. I have one sample  with 0 for all 68
features (in the matrix created by classify_fl), but it was predicted as
dFL. Does it make sense for you?

Baoyan

**Comment by bbai89** [comment]
sorry for so many questions. Can I change the parameters in the
randomForest function?

randomForest(formula = FL.class ~ ., data = train_dat_common,
 importance = TRUE, mtry = 3, maxnodes = 25, ntree = 500,      na.action =
"drop")

I tried, but I do not know how to get train_dat_common. Thanks again!

**Comment by Kdreval** [comment]
No worries about the questions - always glad to help!
Regarding the sample with 0 features in your data, from the classifier point of view, the assigned label makes sense - since there are no cFL features the model will revert to dFL label. But from the data point of view, I would scrutinize that sample for the QC and see if it is worth having this sample in the cohort, and may potentially drop it. I would look at the X coverage, the proportion of targets (or genome) with 30X coverage, proportion of duplicate reads, tumor purity, total mutation burden and number of coding and lymphoma-related SSM, mean VAF of those mutations, and how that compares to the rest of the cohort.
Regarding the parameters in the random forest object, this is a "locked in" representation of how the model was developed in the first place. It is there for tracking purposes and can't be modified unless you train a different model separately - then these parameters can be adjusted.

**Comment by bbai89** [comment]
Thank you very much for the explanation. It makes sense.

I wonder if we are going to train a different model, would it be ok to
include your training data?  I am thinking of combining our sequencing data
to train a new model and see whether the accuracy of classify_fl on our
data can be improved.

Baoyan

**Comment by Kdreval** [comment]
Absolutely, for sure - please feel free to include the data. It can be pulled from the supplement of the paper (if you are comparing to exomes and looking for coding variants only) or I can also suggest to explore the [GAMBRL.data package](https://morinlab.github.io/GAMBLR.data/) which also contains the mutations in aSHM regions in addition to the paper supplement.

**Comment by bbai89** [comment]
Hi Kostiantyn,

I have to work with other projects and am now back to your classification.
Sorry for disturbing you again!

I tried to understand how you define hotspots for a few genes (MEF2B, EZH2,
MYD88, STAT6, FOXO1, CREBBPinKATdomain). I pulled out
"GAMBLR.data::hotspots_annotations". Is this a list of regions where you
define them as hotspots (see below)?  In addition,  I do not understand
what MAX_COORD means.

Thank you for your time!

Regards,
Baoyan

Start_Position Chromosome hot_spot
3786691 16 TRUE
3786692 16 TRUE
3786693 16 TRUE
3786694 16 TRUE
3786695 16 TRUE
3786696 16 TRUE
3786697 16 TRUE
3786698 16 TRUE
3786699 16 TRUE
3786700 16 TRUE
3786701 16 TRUE
3786702 16 TRUE
3786703 16 TRUE
3786704 16 TRUE
3786705 16 TRUE
3786706 16 TRUE
3786707 16 TRUE
3786708 16 TRUE
3786709 16 TRUE
3786710 16 TRUE
3786711 16 TRUE
3786712 16 TRUE
3786713 16 TRUE
3786714 16 TRUE
3786715 16 TRUE
3786716 16 TRUE
3786717 16 TRUE
3786718 16 TRUE
3786719 16 TRUE
3786720 16 TRUE
3786721 16 TRUE
3786722 16 TRUE
3786723 16 TRUE
3786724 16 TRUE
3786725 16 TRUE
3786726 16 TRUE
3786727 16 TRUE
3786728 16 TRUE
3786729 16 TRUE
3786730 16 TRUE
3786731 16 TRUE
3786732 16 TRUE
3786733 16 TRUE
3786734 16 TRUE
3786735 16 TRUE
3786736 16 TRUE
3786737 16 TRUE
3786738 16 TRUE
3786739 16 TRUE
3786740 16 TRUE
3788605 16 TRUE
3788606 16 TRUE
3788607 16 TRUE
3788608 16 TRUE
3788609 16 TRUE
3788610 16 TRUE
3788611 16 TRUE
3788612 16 TRUE
3788613 16 TRUE
3788614 16 TRUE
3788615 16 TRUE
3788616 16 TRUE
3788617 16 TRUE
3788618 16 TRUE
19260033 19 TRUE
19260034 19 TRUE
19260035 19 TRUE
19260036 19 TRUE
19260037 19 TRUE
19260038 19 TRUE
19260039 19 TRUE
19260040 19 TRUE
19260041 19 TRUE
19260042 19 TRUE
19260043 19 TRUE
19260044 19 TRUE
19260045 19 TRUE
19260046 19 TRUE
19260047 19 TRUE
19260048 19 TRUE
19260049 19 TRUE
19260050 19 TRUE
19260051 19 TRUE
19260052 19 TRUE
19260053 19 TRUE
19260054 19 TRUE
19260055 19 TRUE
19260056 19 TRUE
19260057 19 TRUE
19260058 19 TRUE
19260059 19 TRUE
19260060 19 TRUE
19260061 19 TRUE
19260062 19 TRUE
19260063 19 TRUE
19260064 19 TRUE
19260065 19 TRUE
19260066 19 TRUE
38182032 3 TRUE
38182032 3 TRUE
38182641 3 TRUE
38182641 3 TRUE
41240273 13 TRUE
41240274 13 TRUE
41240275 13 TRUE
41240276 13 TRUE
41240277 13 TRUE
41240278 13 TRUE
41240279 13 TRUE
41240280 13 TRUE
41240281 13 TRUE
41240282 13 TRUE
41240283 13 TRUE
41240284 13 TRUE
41240285 13 TRUE
41240286 13 TRUE
41240287 13 TRUE
41240288 13 TRUE
41240289 13 TRUE
41240290 13 TRUE
41240291 13 TRUE
41240292 13 TRUE
41240293 13 TRUE
41240294 13 TRUE
57496654 12 TRUE
57496655 12 TRUE
57496656 12 TRUE
57496657 12 TRUE
57496658 12 TRUE
57496659 12 TRUE
57496660 12 TRUE
57496661 12 TRUE
57496662 12 TRUE
57496654 12 TRUE
57496655 12 TRUE
57496656 12 TRUE
57496657 12 TRUE
57496658 12 TRUE
57496659 12 TRUE
57496660 12 TRUE
57496661 12 TRUE
57496662 12 TRUE
57496663 12 TRUE
57496664 12 TRUE
57496665 12 TRUE
57496666 12 TRUE
57496667 12 TRUE
57496668 12 TRUE
57496669 12 TRUE
57496670 12 TRUE
57496671 12 TRUE
148508727 7 TRUE
148508728 7 TRUE
148508729 7 TRUE
148508730 7 TRUE
148508731 7 TRUE
148508732 7 TRUE
148508733 7 TRUE
148508734 7 TRUE
148508735 7 TRUE
148508736 7 TRUE
148508737 7 TRUE
148508738 7 TRUE
148508739 7 TRUE
148508740 7 TRUE
148508741 7 TRUE
148508742 7 TRUE
148508743 7 TRUE
148508744 7 TRUE
148508745 7 TRUE

**Comment by Kdreval** [comment]
Yes, this is a list of regions - they are defined by the tool OncodriveCLUSTL: https://pubmed.ncbi.nlm.nih.gov/31228182/
We run it on the mix of the FL and DLBCL to detect hotspot regions.
In addition to the OncodriveCLUSTL output, the hotspots are also reviewed - using the review_hotspots function from GAMBLR.data: https://github.com/morinlab/GAMBLR.data/blob/master/R/review_hotspots.R

**Comment by bbai89** [comment]
Thank you for the clarification.

There are some redundancies in this list, it may be a mistake? So any
mutations falling in these regions will be defined as hotspots?

**Comment by Kdreval** [comment]
Right, any mutation in these regions will be considered as hotspots - and possible redundancies (e.g. `57496662 12 TRUE`) are coming from different combinations of sample sets (for example only FL, only DLBCL, FL+DLBCL together etc)

**Comment by bbai89** [comment]
Thanks a lot for your quick reply! Have a nice 1th May!
